### DISCUSSION TITLE
Gif Framerate and weird artifacts (dots) when converting png sequence to .gif

### DISCUSSION THREAD

**Discussion by AquaCylinder** [ImageMagick/ImageMagick#2214] [Help] [answered]
Hi so i made a PS script to make it easier for me to convert a png sequence into a gif by using these two very simple lines of code:
`cd (Get-Clipboard).Replace('"',"")`
`convert *.png -delay 1x30 -loop 0 gif.gif`
But first of all it doesn't run on 30Fps it's just stuck and 10Fps whatever i change in the -delay extension, maybe i just didn't understand it correctly but still doesn't add up.
Secondly i have dots in the final render that of course shouldn't be there, even though i have used this simple script a lot the last couple of weeks. This is the final render that came out at the end:
![gif](https://user-images.githubusercontent.com/56627792/86052824-e085af00-ba57-11ea-8a77-9c94a5156e19.gif)
Thanks for the support!

**Comment by fmw42** [comment]
Put -delay X before reading the input PNGs. 

Note: GIFs do not support 8-bit transparency. They only support binary transparency. So if your PNGs have transparency that is not binary, but 8-bit gray values, it will be converted to binary in the GIFs. So that will cause aliasing artifacts.

**Reply by AquaCylinder** [reply]
thank you, but the dots? Those are in a location where there is no transparency does that still effect it or?

**Reply by fmw42** [reply]
Post a zip file of your input images, so I can see what is happening.

**Reply by AquaCylinder** [reply]
https://drive.google.com/file/d/1gTOMkifplnvz4YGCJvfRd29-thwgTKb6/view?usp=sharing
Thank you btw!

**Comment by fmw42** [comment]
I do not get any dots when I do it using IM 6.9.11.20 Q16 Mac OSX with libpng 1.6.37.  What is your IM version and platform/OS? What is your version of libpng? 

`convert -delay 10 *.png -layers optimize -loop 0 anim.gif`


![anim](https://user-images.githubusercontent.com/13458218/86062990-b95ebb80-ba1e-11ea-94ff-527ed80bc420.gif)

**Reply by AquaCylinder** [reply]
so i have IM 7.0.10-19 Q16, windows 10. I don't know what libpng version i have as the command `identify -list format | grep PNG` gave me this error `'grep' is not recognized as an internal or external command, operable program or batch file`, i will try your command to check if i get the same result shortly,
Edit: it did remove the dots but there did seam to be another artifact, but the sollution was to just use +dither before writing the gif XD

**Comment by snibgo** [comment, marked as answer]
I think the white dots are caused by (a) the variation in the near-black colours, (b) the reduction to 256 colours and (c) the dithering by the colour-reduction algorithm.

I suggest inserting "+dither" before writing the GIF.

**Reply by AquaCylinder** [reply]
Ah yes that worked, i thought it that that would be it but didn't really know how to implement it. it looks awesome without the dithering and the 30Fps timings! Thank you!
![gif dithered](https://user-images.githubusercontent.com/56627792/86115646-87a42e00-bacc-11ea-80f5-f1a6227f1650.gif)

**Comment by snibgo** [comment]
```
it looks awesome with the dithering and the 30Fps timings!
```
I'm glad that works. However, "+dither" prevents dithering, which was my intention. See https://www.imagemagick.org/script/command-line-options.php#dither

For example, if the input is a solid 5% gray and we remap to the two colours black and white with dithering, the output will have mostly black pixels but one in 20 will be white.

But if we prevent dithering, all output pixels will be black.

**Reply by AquaCylinder** [reply]
gotya, makes sense, thank you for the explenation
